packages/mobile/src/fiatExchanges/ProviderOptionsScreen.tsx

```diff
@@ -91,6 +116,21 @@ function ProviderOptionsScreen({ route, navigation }: Props) {
     })
   }, [])
 
+  useAsync(async () => {
+    try {
+      const locationData = await fetchLocationFromIpAddress()
+      if (!locationData) {
+        throw Error('Url returned from service is invalid')
+      }
+
+      const { alpha2, state } = locationData
+      setUserLocation({ country: alpha2, state })
+    } catch (error) {
+      const country = countryCallingCode ? getRegionCodeFromCountryCode(countryCallingCode) : null
+      setUserLocation({ country, state: null })
+    }
```

Comment:
`useAsync` should not be used this way because it will lead to cases where `setUserLocation` is called after the component has been unmounted.

Instead, we should use the result object returned by `useAsync` and remove the `useState` introduced here.